polkadot-parachain/src/command.rs

```diff
@@ -774,92 +900,126 @@ pub fn run() -> Result<()> {
 				<E as NativeExecutionDispatch>::ExtendHostFunctions,
 			>;
 
-			let info_provider = timestamp_with_aura_info(6000);
+			let info_provider = timestamp_with_aura_info::<Block>(6000);
 
 			match runner.config().chain_spec.runtime() {
-				Runtime::AssetHubKusama => runner.async_run(|_| {
-					Ok((
-						cmd.run::<Block, HostFunctionsOf<AssetHubKusamaExecutor>, _>(Some(
-							info_provider,
-						)),
-						task_manager,
-					))
-				}),
-				Runtime::AssetHubWestend => runner.async_run(|_| {
-					Ok((
-						cmd.run::<Block, HostFunctionsOf<AssetHubWestendExecutor>, _>(Some(
-							info_provider,
-						)),
-						task_manager,
-					))
-				}),
-				Runtime::AssetHubPolkadot => runner.async_run(|_| {
-					Ok((
-						cmd.run::<Block, HostFunctionsOf<AssetHubPolkadotRuntimeExecutor>, _>(Some(
-							info_provider,
-						)),
-						task_manager,
-					))
-				}),
-				Runtime::CollectivesPolkadot | Runtime::CollectivesWestend =>
+				Runtime::AssetHubKusama => ensure_feature!(
+					"asset-hub-kusama-runtime",
+					runner.async_run(|_| {
+						Ok((
+							cmd.run::<Block, HostFunctionsOf<AssetHubKusamaExecutor>, _>(Some(
+								info_provider,
+							)),
+							task_manager,
+						))
+					})
+				),
+				Runtime::AssetHubWestend => ensure_feature!(
+					"asset-hub-westend-runtime",
+					runner.async_run(|_| {
+						Ok((
+							cmd.run::<Block, HostFunctionsOf<AssetHubWestendExecutor>, _>(Some(
+								info_provider,
+							)),
+							task_manager,
+						))
+					})
+				),
+				Runtime::AssetHubPolkadot => ensure_feature!(
+					"asset-hub-polkadot-runtime",
+					runner.async_run(|_| {
+						Ok((
+							cmd.run::<Block, HostFunctionsOf<AssetHubPolkadotRuntimeExecutor>, _>(
+								Some(info_provider),
+							),
+							task_manager,
+						))
+					})
+				),
+				Runtime::CollectivesPolkadot | Runtime::CollectivesWestend => ensure_feature!(
+					"collectives-runtime",
 					runner.async_run(|_| {
 						Ok((
 							cmd.run::<Block, HostFunctionsOf<CollectivesPolkadotRuntimeExecutor>, _>(Some(info_provider)),
 							task_manager,
 						))
-					}),
-				Runtime::BridgeHub(bridge_hub_runtime_type) =>
+					})
+				),
+				Runtime::BridgeHub(bridge_hub_runtime_type) => ensure_feature!(
+					"bridge-hub-runtimes",
 					match bridge_hub_runtime_type {
 						chain_spec::bridge_hubs::BridgeHubRuntimeType::Polkadot |
 						chain_spec::bridge_hubs::BridgeHubRuntimeType::PolkadotLocal |
-						chain_spec::bridge_hubs::BridgeHubRuntimeType::PolkadotDevelopment =>
+						chain_spec::bridge_hubs::BridgeHubRuntimeType::PolkadotDevelopment => ensure_feature!(
```

Comment:
unnecessary check of `bridge-hub-runtimes` feature. It has been checked before `match` keyword.